src/openms/source/CHEMISTRY/ISOTOPEDISTRIBUTION/CoarseIsotopePatternGenerator.cpp

```diff
@@ -179,6 +179,71 @@ namespace OpenMS
     return result;
   }
 
+  // static
+  IsotopeDistribution CoarseIsotopePatternGenerator::approximateFromPeptideWeight(double mass, int num_peaks)
+  {
+    IsotopeDistribution result;
+    result.resize(num_peaks);
+
+    // lambda * mass. Lambda is the parameter for Poisson distribution. Value (1/1800) taken from Bellew et al
+    double factor = mass / 1800.0;
+
+    // values of (m * lambda) ^ k
+    double curr_power = 1.0;
+
+    // values of k!
+    UInt curr_factorial = 1;
+
+    // for k=0, non-normalized value is always 1
+    result[0] = Peak1D(mass, 1.0);
+
+    float curr_intensity;
+    for (UInt k = 1; k < num_peaks; ++k)
+    {
+      curr_power *= factor;
+      curr_factorial *= k;
+      curr_intensity = curr_power / curr_factorial;
+      result[k] = Peak1D(mass + (k * OpenMS::Constants::NEUTRON_MASS_U),
+        std::isinf(curr_intensity) ? 0 : curr_intensity);// at some point, curr_intensity will become too small for float (which is the intensity type)
+    }
+
+    result.renormalize();
+
+    return result;
+  }
+
+  std::vector<float> CoarseIsotopePatternGenerator::approximateIntensities(double mass, int num_peaks)
```

Comment:
`num_peaks` should be unsigned, otherwise calling it with `-1` and funny things will happen :)
Maybe even make it `uint8_t`, if we feel that more than 255 peaks are never needed (and that many peaks are only useful for very large molecules, for which this approximation does not work anymore)

Reply:
uint would be better, agree.

But maybe don't restrict it to 8 bit
1. one might actually want to use that method to compare the error between the approximation and the exact calculation for larger molecules
2. I don't know if we properly wrap the 8bit uint type in python. Anyone knows?

so going for uint would be a "practical" solution :)

Reply:
Changed it to UInt